### PR TITLE
Refactor Input Component

### PR DESCRIPTION
**Pull request by Monte9**
## Major Breaking changes 

Please see the [`/example` app](https://github.com/react-native-training/react-native-elements/blob/button_v1/example/src/views/input_home.js) for the above implementation.

### Really simple and useful API:
```
Input.propTypes = {
  containerStyle: ViewPropTypes.style,

  icon: PropTypes.object,
  iconContainerStyle: ViewPropTypes.style,

  inputStyle: PropTypes.object,

  displayError: PropTypes.bool,
  errorStyle: PropTypes.object,
  errorMessage: PropTypes.string,
};
```

Also handles 3 TextInput functions:
```
this.input.focus();
this.input.blur();
this.input.clear();
```

### CODE REVIEW

File: src/input/Input.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import PropTypes from 'prop-types';
```
Comment (Monte9) [inline src/input/Input.js]: this is the file that needs to be reviewed.

**Comment by Monte9** [comment]
Also here are some cool UI designs to port over using RNE: https://www.pinterest.com/timoa/mobile-ui-logins/

$100 🤑 for anyone wanting to contribute and create an entire screen

**Comment by iRoachie** [comment]
@Monte9 Maybe select some, out of the bunch?

**Comment by Monte9** [comment]
@iRoachie that's a good point. Here are the ones that I think looks good/can be a bit challenging to implement. 
1. https://dribbble.com/shots/2186079-Sign-In-Screen/attachments/403389
2. https://dribbble.com/shots/2782171-Mobile-App-SIGN-IN-screen-prototype/attachments/568100
3. https://dribbble.com/shots/2580030-Sign-up/attachments/512793
4. https://www.inspirationde.com/image/70204/

**Comment by iRoachie** [comment]
@Monte9 Is it only the text fields you're interested in?

**Comment by Monte9** [comment]
Ideally, the whole view. Any of those can easily be recreated with the refactored Button component, Input component, bg image, and flex. 

For instance, here's one I created:
simulator screen shot jul 15 2017 9 44 42 pm

**Comment by iRoachie** [comment]
@Monte9 Oh so screens to put in the showcase? I can make them, just wondering what was reasoning behind it.

**Comment by Monte9** [comment]
@iRoachie exactly. The idea is to help dev's bootstrap their RN app can by giving them an "unfair" advance as they can literally copy a "screen" from our showcase app and built on top of it. 

And on the flip side, any dev that creates these screens and contributes back to RNE, will be getting paid $50/screen for it. So it's a win-win. 

It would be awesome if you can create the first screen and kick this idea off. 😉

**Comment by iRoachie** [comment]
Will take a jab after release 0.14

**Comment by Monte9** [comment]
@xavier-villelegier @martinezguillaume is this something you guys would be interested in? ^^

**Comment by xavier-villelegier** [comment]
@Monte9 Sure, i'll try to look into it this week. It can be a good showcase of what can be done with RNE. On which branch should we be based to have those new components (`input_v1`/`button_v1`) ? The idea is to put these screens in the new example app ?

**Comment by Monte9** [comment]
Alright, the `Input v1` API looks pretty solid to me. I was able to create a pretty nice looking input field with it for login. (see update image above)

Also there are lots of opportunities to help out with this if anyone is interested. Please see this: https://github.com/react-native-training/react-native-elements/projects/4

Merging this PR into `v1`

**Comment by Monte9** [comment]
@xavier-villelegier all the latest changes are on the `v1` branch. You can branch off of that as it has the new `Button` & `Input` components.

Also the idea is to basically create entire screens for the app (login view/sign up view/feed view/profile view) using `RNE v1` so that anyone looking for UI ideas and literally copy and paste the code to get started. Plus it will show them how useful the RNE library is and show them how to leverage the library the right way. 

You can see all the available tasks here: https://github.com/react-native-training/react-native-elements/projects/4

/cc @iRoachie ^^

**Comment by iRoachie** [comment]
@Monte9 Gotcha will chip in after the release. Release notes are fun :D

**Pull request merged**